### PR TITLE
feat: Add support for Starlark float literals

### PR DESCRIPTION
**Pull request by timothyg-stripe**
# Checklist

- [x] I have filed an issue about this change and discussed potential changes with the maintainers.
- [x] I have received the approval from the maintainers to make this change.
- [x] This is not a stylistic, refactoring, or cleanup change.

Please note that the maintainers will not be reviewing this change until all checkboxes are ticked. See 
the [Contributions](https://github.com/bazelbuild/intellij#contributions) section in the README for more 
details.

# Discussion thread for this change

Issue number: #5677

# Description of this change

Add support for Starlark floating-point literals. Also add `float` as a built-in function, and consider float values primitives in the debugger.

Before:
not highlighted

After:
highlighted correctly

Much of the lexer implementation is ported from [starlark-go](https://github.com/google/starlark-go/blob/556fd59b42f68a2fb1f84957741b72811c714e51/syntax/scan.go#L909)

### CODE REVIEW

File: base/tests/unittests/com/google/idea/blaze/base/lang/buildfile/lexer/BlazeLexerTest.java [hunk 1]
```diff
@@ -63,28 +63,28 @@ public void testBasics4() throws Exception {
...
+    assertEquals("invalid float constant: 1.23E", lastError);
 
+    assertEquals("FLOAT(1.23E-10)", values(tokens("1.23E-10")));
     assertEquals("DOT INT(123)", values(tokens(". 123")));
-    assertEquals("DOT INT(123)", values(tokens(".123")));
+    assertEquals("FLOAT(0.123)", values(tokens(".123")));
     assertEquals("DOT IDENTIFIER(abc)", values(tokens(".abc")));
 
-    assertEquals("IDENTIFIER(foo) DOT INT(123)", values(tokens("foo.123")));
+    assertEquals("IDENTIFIER(foo) FLOAT(0.123)", values(tokens("foo.123")));
```
Comment (blorente) [inline base/tests/unittests/com/google/idea/blaze/base/lang/buildfile/lexer/BlazeLexerTest.java:87]: I'm not sure the spec covers this, but I think an identifier that ends in numbers should still parse the same way:

```suggestion
    assertEquals("IDENTIFIER(foo) FLOAT(0.123)", values(tokens("foo.123")));
    assertEquals("IDENTIFIER(foo456) FLOAT(0.123)", values(tokens("foo567.123")));
```

**Review by blorente — APPROVED** [review]
The changes in general look good, thanks for working on them! Added a corner case I'd like to test. If that passes, I think we're good to go.

File: base/tests/unittests/com/google/idea/blaze/base/lang/buildfile/lexer/BlazeLexerTest.java (hunk 1 above)
Comment (timothyg-stripe) [reply base/tests/unittests/com/google/idea/blaze/base/lang/buildfile/lexer/BlazeLexerTest.java:87]: The spec says:

> Each token is formed from the longest sequence of characters that would form a valid token of each kind.

(i.e., the "maximal munch") so it does cover this.

Test added, passes :)
